### PR TITLE
Add a switch to turn on/off mt analysis

### PR DESCRIPTION
**Pull request by ramprasadn**
## PR checklist

See title

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/raredisease/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/raredisease _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR> -stub`).
- [x] Ensure the test suite passes (`nextflow run . -profile test2samples,docker --outdir <OUTDIR> -stub`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: workflows/raredisease.nf [hunk 1]
```diff
@@ -403,19 +408,28 @@ workflow RAREDISEASE {
...
+                .set { ch_merged_tbi }
+
+            ANNOTATE_SNVS.out.vcf_ann
+                .concat(ANALYSE_MT.out.vcf)
+                .groupTuple()
+                .set { ch_merged_vcf }
+
+            ch_merged_vcf.join(ch_merged_tbi).set {ch_concat_in}
+
+            BCFTOOLS_CONCAT (ch_concat_in)
```
Comment (jemten) [inline workflows/raredisease.nf:427]: This step could be troublesome. We need to either remove the existing MT variants from the vcf produced by DV/GLnexus or run bcftools concat with the '-a' option, which I can't see that we do.

File: nextflow.config [hunk 2]
```diff
@@ -24,6 +24,7 @@ params {
     analysis_type              = 'wgs'
     annotate_snv_switch        = true
     annotate_sv_switch         = false
+    annotate_mt_switch         = true
```
Comment (jemten) [inline nextflow.config:27]: I would rename this parameter. We're not only annotating MT variants, we are also aligning and calling. Perhaps `dedicated_mt_workflow`?

**Review by jemten — CHANGES_REQUESTED** [review]
Some comments but nothing major. The merging could use some more work but we can also do that in another PR.

File: workflows/raredisease.nf (hunk 1 above)
Comment (ramprasadn) [reply workflows/raredisease.nf:427]: We are removing the mitochondrial variants with GATK's selectvariants in the call snv subworkflow

File: workflows/raredisease.nf [hunk 3]
```diff
@@ -403,19 +408,28 @@ workflow RAREDISEASE {
...
+
+            ANNOTATE_SNVS.out.tbi
+                .concat(ANALYSE_MT.out.tbi)
+                .groupTuple()
+                .set { ch_merged_tbi }
+
+            ANNOTATE_SNVS.out.vcf_ann
+                .concat(ANALYSE_MT.out.vcf)
+                .groupTuple()
+                .set { ch_merged_vcf }
```
[the remark this answers is not in the document]
Comment (ramprasadn) [reply workflows/raredisease.nf:423]: makes sense

File: nextflow.config (hunk 2 above)
Comment (ramprasadn) [reply nextflow.config:27]: good idea 👍🏻

File: workflows/raredisease.nf (hunk 1 above)
Comment (ramprasadn) [reply workflows/raredisease.nf:427]: But I should move it to the raredisease subworkflow and run it just before executing bcftools concat. As it stands now, if one chooses not to run the dedicated mt workflow then the variants from deepvariant will still be removed 😅

Comment (jemten) [reply workflows/raredisease.nf:427]: Ah! Missed that one. 
However, isn't that part always executing? I don't think we should remove the MT variants unless we are running a dedicated workflow.

Comment (jemten) [reply workflows/raredisease.nf:427]: You beat me to it 😆

**Review by jemten — APPROVED** [review]
Looks good! ⭐

**Pull request merged**